app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -174,7 +170,54 @@ public function getCartDataJs($multipage = true)
 
             // Call Bolt create order API
             try {
-                $orderCreationResponse = $this->createBoltOrder($quote, $multipage);
+                /////////////////////////////////////////////////////////////////////////////////
+                // We create a copy of the quote that is immutable by the customer/frontend
+                // Bolt saves this quote to the database at Magento-side order save time.
+                // This assures that the quote saved to Magento matches what is stored on Bolt
+                // Only shipping, tax and discounts can change, and only if the shipping, tax
+                // and discount calculations change on the Magento server
+                ////////////////////////////////////////////////////////////////////////////////
+
+                /*********************************************************/
+                /* Clean up resources that may have previously been saved
+                /* @var Mage_Sales_Model_Quote[] $expired_quotes */
+                $expired_quotes = Mage::getModel('sales/quote')
+                    ->getCollection()
+                    ->addFieldToFilter('parent_quote_id', $quote->getId());
+
+                foreach( $expired_quotes as $expired_quote) {
+                    $expired_quote->delete();
+                }
+                /*********************************************************/
+
+                /* @var Mage_Sales_Model_Quote $immutableQuote */
+                $immutableQuote = Mage::getSingleton('sales/quote');
+
+                if (!$multipage) {
+                    // For the checkout page we want to set the
+                    // billing and shipping, and shipping method at this time.
+                    // For multi-page, we add the addresses during the shipping and tax hook
+                    // and the chosen shipping method at order save time.
+                    $immutableQuote
+                        ->setBillingAddress($quote->getBillingAddress())
+                        ->setShippingAddress($quote->getShippingAddress())
+                        ->getShippingAddress()
+                            ->setShippingMethod($quote->getShippingAddress()->getShippingMethod())
+                            ->save();
+                }
+
+                $immutableQuote
+                    ->setCustomer($quote->getCustomer())
+                    ->merge($quote)
+                    // Generate new increment order id and associate it with current quote, if not already assigned
+                    ->reserveOrderId()
```

Comment:
I don't know if this is an issue or not, but won't this increment the reserveOrderId on every cart or checkout page refresh. I'm not sure if that's an issue, but I know that before I realized that a reserved order id didn't always have a one-to-one mapping with an order entity_id, I would often look at the increment id to gauge how many orders were in the system. I know this is stupid, but I think it would cause some people to think there is something wrong if we drastically increase the gap between increment ids. However, this may not be an issue. Just pointing it out.

Reply:
@johnmccombs1 I think it is helpful for me to review code from the diff view.  It helps me answer questions like this visually and you might find it to be useful so I will provide a link.

https://github.com/BoltApp/bolt-magento1/pull/49/files

Anyway, likely, you are already doing this, but just in case, see above.

Now, to answer your questions, if you look at the diff in the files, you will notice that `reservedOrderId()` is called exactly the same number of times at the same point in the order flow as it was before.  Therefore, there is no change to how order Ids are being reserved and thus, there is no difference in what it will appear like in the underlying database.

Then we must ask the question, "is the preceding implementation wrong?"  The answer is that the proceeding implementation is correct, as is its continued use.  This is because calling this function will only generate a new reserved order id if one is not already assigned to the quote and the id is not in use by a current order.  You can look this up in the Magento source code.

However, perhaps your observation would be correct if the quote was not saved after the call to reserve order id.  It is saved on line 219.